packages/x-charts/src/ChartsOverlay/ChartsLoadingOverlay.tsx

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { styled } from '@mui/material/styles';
+import { useDrawingArea } from '../hooks/useDrawingArea';
+import type { CommonOverlayProps } from './ChartsOverlay';
+
+const StyledText = styled('text')(({ theme }) => ({
+  stroke: 'none',
+  fill: theme.palette.text.primary,
+  shapeRendering: 'crispEdges',
+  textAnchor: 'middle',
+  dominantBaseline: 'middle',
+}));
+
+export function ChartsLoadingOverlay(props: CommonOverlayProps) {
+  const { message, ...other } = props;
+  const { top, left, height, width } = useDrawingArea();
+
+  return (
+    <StyledText x={left + width / 2} y={top + height / 2} {...other}>
+      {message ?? 'Loading data ...'}
```

Comment:
I guess we will need locales for charts too.

```suggestion
      {message ?? 'Loading data…'}
```